src/XMakeBuildEngine/BackEnd/Components/Communications/NodeProviderInProc.cs

```diff
@@ -230,7 +230,11 @@ public bool CreateNode(int nodeId, INodePacketFactory factory, NodeConfiguration
 
                 if (_exclusiveOperatingEnvironment)
                 {
+#if FEATURE_NAMED_SYNCHRONIZATION_PRIMITIVES
                     _inProcNodeOwningOperatingEnvironment = new Semaphore(1, 1, "MSBuildIPN_" + Process.GetCurrentProcess().Id);
+#else
+                    _inProcNodeOwningOperatingEnvironment = new Semaphore(1, 1, null);
```

Comment:
I'm concerned that this is not equivalent behavior. On Windows, any call through this codepath in a process will get the same semaphore, but now there's a new semaphore per call--so it can always be entered in the next statement.

Should this be replaced with a static semaphore that gets populated once (per process)?